src/components/PeoplePicker/PeoplePicker.tsx

```diff
@@ -1,113 +1,213 @@
-import React from 'react';
-import { IPersonaProps } from '@fluentui/react/lib/Persona';
-import { IBasePickerSuggestionsProps, NormalPeoplePicker } from '@fluentui/react/lib/Pickers';
-import { people } from '@fluentui/example-data';
+import { FunctionComponent, useEffect, useRef, useState } from "react";
+import { IPersonaProps } from "@fluentui/react/lib/Persona";
+import {
+  IBasePicker,
+  IBasePickerSuggestionsProps,
+  NormalPeoplePicker,
+} from "@fluentui/react/lib/Pickers";
+import { people } from "@fluentui/example-data";
+import { spWebContext } from "../../providers/SPWebContext";
+import { IPeoplePickerEntity } from "@pnp/sp/profiles";
+
+// TODO: Add a way to show as input needed/corrected
 
 const suggestionProps: IBasePickerSuggestionsProps = {
-  suggestionsHeaderText: 'Suggested People',
-  mostRecentlyUsedHeaderText: 'Suggested Contacts',
-  noResultsFoundText: 'No results found',
-  loadingText: 'Loading',
+  suggestionsHeaderText: "Suggested People",
+  mostRecentlyUsedHeaderText: "Suggested Contacts",
+  noResultsFoundText: "No results found",
+  loadingText: "Loading",
   showRemoveButtons: true,
-  suggestionsAvailableAlertText: 'People Picker Suggestions available',
-  suggestionsContainerAriaLabel: 'Suggested contacts',
+  suggestionsAvailableAlertText: "People Picker Suggestions available",
+  suggestionsContainerAriaLabel: "Suggested contacts",
 };
 
-export const PeoplePicker: React.FunctionComponent<any> = (props) => {
-  const [currentSelectedItems, setCurrentSelectedItems] = React.useState<IPersonaProps[]>([]);
-  const [peopleList] = React.useState<IPersonaProps[]>(people);
+export interface SPPersona extends IPersonaProps {
+  AccountName?: string;
+  Department?: string;
+  Email?: string;
+  SPUserId?: number;
+}
+
+interface IPeoplePickerProps {
+  /** Required - The text used to label this people picker for screenreaders */
+  ariaLabel: string;
+  /** Optional - The people to pre-populate the People Picker with */
+  defaultValue?: SPPersona[];
+  readOnly?: boolean;
+  required?: boolean;
+  /** Optional - Limit the People Picker to only allow selection of specific number -- Defaults to 1 */
+  itemLimit?: number;
+  updatePeople: (p: SPPersona[]) => void;
+}
+
+export const PeoplePicker: FunctionComponent<IPeoplePickerProps> = (props) => {
+  const [currentSelectedItems, setCurrentSelectedItems] = useState<
+    IPersonaProps[]
+  >([]);
+  const [peopleList] = useState<IPersonaProps[]>(people);
 
-  const picker = React.useRef(null);
+  useEffect(() => {
+    let personas: SPPersona[] = [];
+    if (props.defaultValue) {
+      personas = [...props.defaultValue];
+    }
+    setCurrentSelectedItems(personas);
+  }, [props.defaultValue]);
 
-  const onFilterChanged = (
+  const picker = useRef<IBasePicker<IPersonaProps>>(null);
```

Comment:
You don't actually need this. It's an optional parameter for the picker, and we're not using it to manipulate anything in our code.

Reply:
Created #16 with comment to look at previous implementations and why it was setting focus leveraging this after update